### PR TITLE
use dynamic config building in CI

### PR DESCRIPTION
**Pull request by geoknee**
This works by having a small workflow run, compute the list of chain ids to run genesis validation checks on, and then modify a subsequent config file which circleci then runs.

See https://circleci.com/docs/dynamic-config/ 

This is preferable because the validation checks take about 6 minutes each and need to be sandboxed from each other (they each require the monorepo to be checked out at a different commit). 

⚠️ goes into #489

### CODE REVIEW

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      - checkout
+      - install-just
+      - install-foundry
+      - run:
+          name: run validation checks
+          command: just promotion-test
+      - slack/notify:
+          event: always
+          custom: "{\n  \"blocks\": [\n    {\n      \"type\": \"section\",\n      \"text\": {\n        \"type\": \"mrkdwn\",\n        \"text\": \":control_knobs: The daily report on standard candidate chains is ready!\"\n      }\n    },\n    {\n      \"type\": \"divider\"\n    },\n    {\n      \"type\": \"context\",\n      \"elements\": [\n        {\n          \"type\": \"mrkdwn\",\n          \"text\": \"\U0001F440 Click the link to see what validation checks would fail if all candidate chains were promoted to standard (and exclusions removed).\"\n        }\n      ]\n    }\n  ]\n}\n"
+          channel: C07GZVCCUS0 # to slack channel `notify-superchain-promotion-failures`
```
Comment (vdamle) [inline .circleci/continue_config.yml:156]:
```suggestion
          channel: C07GZVCCUS0 # to slack channel `notify-superchain-promotion-reports`
```

File: .circleci/continue_config.yml [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    steps:
+      - checkout
+      - install-just
+      - install-foundry
+      - run:
+          name: run validation checks
+          command: just promotion-test
+      - slack/notify:
+          event: always
+          custom: "{\n  \"blocks\": [\n    {\n      \"type\": \"section\",\n      \"text\": {\n        \"type\": \"mrkdwn\",\n        \"text\": \":control_knobs: The daily report on standard candidate chains is ready!\"\n      }\n    },\n    {\n      \"type\": \"divider\"\n    },\n    {\n      \"type\": \"context\",\n      \"elements\": [\n        {\n          \"type\": \"mrkdwn\",\n          \"text\": \"\U0001F440 Click the link to see what validation checks would fail if all candidate chains were promoted to standard (and exclusions removed).\"\n        }\n      ]\n    }\n  ]\n}\n"
```
Comment (vdamle) [inline .circleci/continue_config.yml:155]: Kudos to figuring out this format! 👏🏼

File: .circleci/continue_config.yml [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+          filters:
+            branches:
+              only:
+                - main
+  pr-checks:
+    jobs:
+      - golang-validate-genesis-allocs:
+          matrix:
+            parameters:
+              # This value will be replaced by the generate_test_config.sh script
```
Comment (vdamle) [inline .circleci/continue_config.yml:233]: Is below the fallback list, if `targetList` is empty in `generate_test_config.sh`?

**Review by vdamle — APPROVED** [review]
Leaving approved, noted a small question in the comment.

**Comment by geoknee** [comment]
To help us set a "required check" rule on github, I added a statically named "all ok" job which uses a fan-in pattern, and depends on all of the dynamically named jobs.
Screenshot 2024-08-30 at 15 05 49

**Pull request merged**